### PR TITLE
BUG #19860 Corrected use of mixed indexes with .at

### LINKED ISSUE #19860: BUG?: .at not working on object indexes containing some integers
Version 0.22.0

#### Problem description
Using the .at - Method on an Index which contains Integers as well as str/objects raises an Error. This used to be possible using the ``.get_value()``-Method. As ``.at`` is the designated successor (#15269) the same behaviour should be supported.
I also noticed that ``.get_value`` is approx. twice as fast as ``.at``. Is there a specific reason to stick with ``.at``?  (see again #15269 for a speed comparison)

#### Code Sample
```python
import pandas as pd
import numpy as np

data = np.random.randn(10, 5)
df = pd.DataFrame(data, columns=['a', 'b', 'c', 1, 2])
df.at[0, 1]
```

Raises:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/thielc/anaconda3/lib/python3.6/site-packages/pandas/core/indexing.py", line 1868, in __getitem__
    key = self._convert_key(key)
  File "/home/thielc/anaconda3/lib/python3.6/site-packages/pandas/core/indexing.py", line 1915, in _convert_key
    raise ValueError("At based indexing on an non-integer "
ValueError: At based indexing on an non-integer index can only have non-integer indexers
```

### PR DESCRIPTION
**Pull request by mariuspot**
`.at` incorrectly disallowed the use of integer indexes when a mixed index was used
```
s = Series([1,2,3,4,5],index=['a','b','c',1,2])
s.at['a'] # returns 1
s.at[1]   # returns 4
s.at[4]   # raises KeyError, doesn't do fallback indexing
```

Made sure fallback indexing doesn't happen on mixed indexes

- [X] closes #19860
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexing/test_indexing.py [hunk 1]
```diff
@@ -666,6 +666,20 @@ def test_index_type_coercion(self):
                     idxr(s2)['0'] = 0
                     assert s2.index.is_object()
 
+    @pytest.mark.parametrize("index,val", [
```
Comment (jreback) [inline pandas/tests/indexing/test_indexing.py]: can you add a comment with the issue number

Comment (jreback) [reply pandas/tests/indexing/test_indexing.py]: these tests should be in pandas/tests/indexes/test_base (look for other contains tests)

File: pandas/tests/indexing/test_indexing.py [hunk 2]
```diff
@@ -710,6 +724,27 @@ def test_float_index_at_iat(self):
         for i in range(len(s)):
             assert s.iat[i] == i + 1
 
+    def test_mixed_index_at_iat(self):
```
Comment (jreback) [inline pandas/tests/indexing/test_indexing.py]: move these to pandas/tests/indexing/test_scalar.py (see where other ones are)

Comment (jreback) [reply pandas/tests/indexing/test_indexing.py]: add a gh comment with the issue number

Comment (jreback) [reply pandas/tests/indexing/test_indexing.py]: can you add the test from the OP (was on dataframe)

File: pandas/tests/indexing/test_indexing.py [hunk 3]
```diff
@@ -710,6 +724,27 @@ def test_float_index_at_iat(self):
...
+        s = Series([1, 2, 3, 4, 5], index=['a', 'b', 'c', 1, 2])
+        for el, item in s.iteritems():
+            assert s.at[el] == item
+        for i in range(len(s)):
```
Comment (jreback) [inline pandas/tests/indexing/test_indexing.py]: can you check that we have similar tests for .loc (I think so), if not can you add a loc version (inside same test is ok). maybe even just add the .loc tests and move from elsewhere.

you can parameterize these over .at / .loc

**Review by jreback — CHANGES_REQUESTED** [review]
looks pretty good. can you add a whatsnew note in bugfixes indexing for 0.24

**Comment by jreback** [comment]
also there *might* be other .at issues which this closes. pls have a look if you can.

cc @toobaz

File: pandas/tests/indexing/test_indexing.py (hunk 1 above)
Comment (mariuspot) [reply pandas/tests/indexing/test_indexing.py]: `test_index_not_contains` and `test_index_contains` are in the same file a few lines above, should I move these up or create pandas/tests/indexes/test_base and move them there?

**Comment by jreback** [comment]
looks good. can you rebase

**Comment by jreback** [comment]
looks like you picked up some other commits, do this
```git merge upstream/master`` and push again

**Comment by mariuspot** [comment]
`git merge upstream/master` didn't do anything
Not sure what I did now :/
Hope it's ok now

**Comment by mariuspot** [comment]
Ok, I figured out what went wrong with the merge. Should be ok now

File: doc/source/whatsnew/v0.24.0.txt [hunk 4]
```diff
@@ -659,7 +659,11 @@ Indexing
 - Fixed ``DataFrame[np.nan]`` when columns are non-unique (:issue:`21428`)
 - Bug when indexing :class:`DatetimeIndex` with nanosecond resolution dates and timezones (:issue:`11679`)
 - Bug where indexing with a Numpy array containing negative values would mutate the indexer (:issue:`21867`)
+<<<<<<< HEAD
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: hve a rebase issue here

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. looks like some extraneous rebase issue. ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green.

**Comment by mariuspot** [comment]
Looks like travis failed with a segfault on the linker

**Comment by mariuspot** [comment]
@jreback The travis build had a segfault with the linker while setting up the environment. What is the process for that. Is there a way for me to rerun the build or should I make a change and do a new commit?

**Comment by jreback** [comment]
thanks!

**Pull request merged**